src/core/tsi/alts_transport_security.h

```diff
@@ -33,15 +33,20 @@ typedef struct alts_shared_resource {
   gpr_mu mu;
   gpr_cv cv;
   bool is_cq_drained;
+  gpr_refcount refcount;
 } alts_shared_resource;
 
-/* This method returns the address of alts_shared_resource object shared by all
- *    TSI handshakes. */
-alts_shared_resource* alts_get_shared_resource(void);
+/** This method returns mu of alts_shared_resource objet. */
+gpr_mu* grpc_tsi_alts_shared_resource_get_mu();
+
+/** This method ref's alts_shared_resource object. */
+alts_shared_resource* grpc_tsi_alts_shared_resource_ref();
```

Comment:
The suffix `_ref()` seems a bit odd for a function that returns a new instance not previously held by the caller, rather than adding a new ref to an instance that the caller has passed in.  Maybe use `_get()` instead?

Actually, it looks like we expect any given caller to always take one ref, use it for everything they need, and they unref it; there don't seem to be any cases where a given caller will hold more than one ref.  That being the case, how about just calling this `_create()` and changing `_unref()` to `_destroy()`?  That way, the fact that we're actually ref-counting the same object under the hood is abstracted away from the caller.